rfcs/01780_policy_delegation.md

```diff
@@ -0,0 +1,308 @@
+instantiated, it receives the declassification privilege of
+"manifest:https://google.com/oak/manifests/count_people_from_pixels.json ∧
+manifest:https://apple.com/oak/manifests/count_people_from_pixels.json ∧
+wasm_hash:1111111111111111111111111111111111111111".
+
+Note that in this case it receives both privileges (or rather, the conjunction
+between these privileges), plus the privilege for the corresponding Wasm hash
+label, which it would get anyways.
+
+In particular, the conjunctive privilege is quite useful in practice as a
+confidentiality label, because it may be used by the data owner to label data
+that is expected to be declassified by a Wasm module that is trusted by multiple
+parties at the same time, where each party independently publishes a separate
+manifest.
+
+## User-based delegation
+
+So far we have talked about companies being the publisher of such manifest, but
+it is possible for individual users to also publish manifests for specific use
+cases. A user may publish a manifest on a URL hosted on a server they control,
+or even as part of another website, e.g. on github.com. Of course, in the case
+of publishing on a different website, that website is the ultimate root of trust
+of any assertions.
+
+This allows delegation based on existing human-to-human network connections. For
+instance, a person may be an expert in a particular field, and independently
+verify properties of pieces of software and publish them in a manifest. Also it
+is possible for a person publishing a manifest to further delegate to another
+publisher (either company or person) by including the URL of the target manifest
+as a trusted object in their own manifest.
+
+For instance, a user may publish a manifest at the following URL:
+`https://github.com/tiziano88/delegation/count_users_from_pixels.json`, which is
+hosted by `github.com`, but of which the user with handle `tiziano88` is
+expected to be author:
+
+```json
+{
+  "trusted": [
+    {
+      "type": "manifest",
+      "value": "https://google.com/oak/manifests/count_people_from_pixels.json"
+    },
+    {
+      "type": "manifest",
+      "value": "https://facebook.com/oak/manifests/count_people_from_pixels.json"
+    },
+    {
+      "type": "wasm_module",
+      "value": "cf26b96746311908f8fe40340a8b6c1f06f5a1b5"
+    }
+  ]
+}
+```
+
+This manifest essentially delegates to the two manifests identified by the URLs
+(meaning that the union of the objects trusted by them is trusted by the user),
+plus an additional hash that is trusted directly by the user.
+
+## Remote Attestation
+
+Remote attestation is a similar problem to that of trusting Wasm modules with a
+given hash. Different parties may have different opinions of what is considered
+trustworthy, and those opinions in general change over time.
+
+A company may publish a list of hashes corresponding to remote attestation
+quotes of a given system.
+
+For instance, Google may publish a delegation manifest at
```

Comment:
It is not clear to me how this will be used. Will a user attach a label to data so that it can only be processed on enclaves with these measurements?

Reply:
I was thinking more that the client will just use it when verifying a remote attestation quote before sending data. But with https://github.com/project-oak/oak/issues/1473 this may actually be folded into the label system as a label component itself. Anyways how it is used in practice may change over time, once it is published in a standard format.